### PR TITLE
Updates tools to include mapmerge hooks and merge drivers, plus more

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
Ports:
https://github.com/tgstation/tgstation/pull/55658
https://github.com/tgstation/tgstation/pull/55699
https://github.com/tgstation/tgstation/pull/60143
https://github.com/tgstation/tgstation/pull/67489

Enables one-click install for hooks and merge drivers, sets up automatic map merging.
This should be done by all mappers before committing a map, either manually with `"\tools\mapmerge2\Run Before Committing.bat"` or automatically after installing the hooks at `"\tools\hooks\Install.bat"`

This means accidental reverts or conflicts are less likely to happen.

Information on what these tools do can be found in the `readme.md` file within each new folder.

Mapping guidelines updated to reflect new tools, they _**MUST**_ be used to ensure sanity of mappers who end up fixing conflicts.

Removes old changelog tool, that thing is two years old and not coming back.

## Why It's Good For The Game
Automation and good tools for mapping workflows is important, this goes some of the way to making life easier.

## Changelog
Tooling change, not player-focused.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -505,15 +505,14 @@ in the SQL/updates folder.
 * For map edit PRs, we do not accept 'change for the sake of change' remaps, unless you have very good reasoning to do so. Maintainers reserve the right to close your PR if we disagree with your reasoning.
 
 * Map Merge
-  * The following guideline for map merging applies to people who are **NOT** using StrongDMM, please see the StrongDMM section if you are.
-    * You **MUST** run Map Merge prior to opening your PR when updating existing maps to minimize the change differences (even when using third party mapping programs such as FastDMM.)
-    * Failure to run Map Merge on a map after using third party mapping programs (such as FastDMM) greatly increases the risk of the map's key dictionary becoming corrupted by future edits after running map merge. Resolving the corruption issue involves rebuilding the map's key dictionary;
+  * The following guideline for map merging applies to **aALL** mapping contributers.
```
Comment (AffectedArc07) [inline .github/CONTRIBUTING.md:508]:
```suggestion
  * The following guideline for map merging applies to **ALL** mapping contributers.
```
The spelling obsessed man has made a spelling error. Laugh at this user.

Comment (S34NW) [reply .github/CONTRIBUTING.md:508]: oh boy

File: dependencies.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
```
Comment (AffectedArc07) [inline dependencies.sh:1]: We already have `_build_dependencies.sh`. Please integrate your changes into that file.

File: tools/UpdatePaths/Update Paths.bat [hunk 3]
```diff
@@ -0,0 +1,3 @@
+@echo off
```
Comment (AffectedArc07) [inline tools/UpdatePaths/Update Paths.bat:1]: Do not put spaces in bat file names please

File: tools/UpdatePaths/__main__.py [hunk 4]
```diff
@@ -1,17 +1,22 @@
 # A script and syntax for applying path updates to maps.
 import re
 import os
+import sys
 import argparse
```
Comment (AffectedArc07) [inline tools/UpdatePaths/__main__.py:5]:
```suggestion
import re, os, sys, argparse
```

File: tools/UpdatePaths/readme.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+HOW TO USE:
+Drag one of the scripts in the “Scripts” folder onto the .bat file “Update Paths” to open it with the .bat file. Let the script run. 
+
+Use this tool before using mapmerge or opening the map in an editor.
+
+IMPORTANT:
+Please tie the script you are making to the associated PR on github and put it in the scripts folder when you are done.
+
+For example: 67107_Turf_Updates_2
```
Comment (AffectedArc07) [inline tools/UpdatePaths/readme.md:9]: Please make this reference one of our PRs, and update the existing ones please

File: tools/bootstrap/node [hunk 6]
```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
```
Comment (AffectedArc07) [inline tools/bootstrap/node:1]: This is for CBT.


Get this shit out of my codebase.

File: tools/hooks/merge_frontend.py [hunk 7]
```diff
@@ -0,0 +1,169 @@
+# merge_frontend.py
+import sys
+import io
+import os
+import pygit2
+import collections
+import typing
```
Comment (AffectedArc07) [inline tools/hooks/merge_frontend.py:7]:
```suggestion
import sys, io, os, pygit2, collections, typing
```

File: tools/mapmerge2/I Forgot To Map Merge.bat [hunk 8]
```diff
@@ -0,0 +1,7 @@
+@echo off
```
Comment (AffectedArc07) [inline tools/mapmerge2/I Forgot To Map Merge.bat:1]: Same with this bat file

File: tools/mapmerge2/Resolve Map Conflicts.bat [hunk 9]
```diff
@@ -0,0 +1,2 @@
+@call "%~dp0\..\bootstrap\python.bat" -m mapmerge2.merge_driver --posthoc %*
```
Comment (AffectedArc07) [inline tools/mapmerge2/Resolve Map Conflicts.bat:1]: And here

File: tools/mapmerge2/Run Before Committing.bat [hunk 10]
```diff
@@ -0,0 +1,2 @@
+@call "%~dp0\..\bootstrap\python" -m mapmerge2.precommit --use-workdir %*
```
Comment (AffectedArc07) [inline tools/mapmerge2/Run Before Committing.bat:1]: And here

File: tools/mapmerge2/dmm_test.py [hunk 11]
```diff
@@ -0,0 +1,39 @@
+import os
+import sys
```
Comment (AffectedArc07) [inline tools/mapmerge2/dmm_test.py:2]:
```suggestion
import os, sys
```

File: tools/mapmerge2/fixup.py [hunk 12]
```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python3
+import os
+import pygit2
```
Comment (AffectedArc07) [inline tools/mapmerge2/fixup.py:3]:
```suggestion
import os, pygit2
```

File: tools/mapmerge2/merge_driver.py [hunk 13]
```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+import sys
+import collections
```
Comment (AffectedArc07) [inline tools/mapmerge2/merge_driver.py:3]:
```suggestion
import sys, collections
```

File: tools/mapmerge2/precommit.py [hunk 14]
```diff
@@ -1,10 +1,12 @@
 #!/usr/bin/env python3
 import os
+import sys
 import pygit2
```
Comment (AffectedArc07) [inline tools/mapmerge2/precommit.py:4]:
```suggestion
import os, sys, pygit2
```

File: tools/oldChangelog/github_webhook_processor.php [hunk 15]
```diff
@@ -1,355 +0,0 @@
-<?php
```
Comment (AffectedArc07) [inline tools/oldChangelog/github_webhook_processor.php:1]: :salute:

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

File: tools/hooks/python.sh [hunk 16]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+set -e
+if [ "$*" = "-m precommit" ]; then
+	echo "Hooks are being updated..."
+	echo "Details: https://github.com/tgstation/tgstation/pull/55658"
```
Comment (AffectedArc07) [inline tools/hooks/python.sh:5]: Link to this PR

File: tools/hooks/python.sh [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+	if [ "$(uname -o)" = "Msys" ]; then
+		tools/hooks/Install.bat
+	else
+		tools/hooks/install.sh
+	fi
+	echo "---------------"
+	exec tools/hooks/pre-commit.hook
+else
+	echo "tools/hooks/python.sh is replaced by tools/bootstrap/python"
+	echo "Details: https://github.com/tgstation/tgstation/pull/55658"
```
Comment (AffectedArc07) [inline tools/hooks/python.sh:15]: And here

Comment (S34NW) [inline tools/hooks/python.sh:15]:
```suggestion
	echo "Details: https://github.com/ParadiseSS13/Paradise/pull/18034 and https://github.com/tgstation/tgstation/pull/55658"
```

**Review by AffectedArc07 — APPROVED** [review]
May as well

**Pull request merged**